src/utils.rs

```diff
@@ -1075,6 +1080,22 @@ pub fn format_url(url: &str) -> String {
 	}
 }
 
+// Remove tracking query params
+static URL_CLEANER: Lazy<Mutex<UrlCleaner>> = Lazy::new(|| Mutex::new(UrlCleaner::from_embedded_rules().expect("Failed to initialize UrlCleaner")));
+
+pub fn clean_url(url: String) -> String {
+	let is_external_url = match Url::parse(url.as_str()) {
+		Ok(parsed_url) => parsed_url.domain().is_some(),
+		_ => false,
+	};
+	let mut cleaned_url = url.clone();
+	if is_external_url {
+		let cleaner = URL_CLEANER.lock().unwrap();
+		cleaned_url = cleaner.clear_single_url_str(cleaned_url.as_str()).expect("Unable to clean the URL.").as_ref().to_owned();
+	}
+	cleaned_url
+}
```

Comment:
This should handle the failed cases better - specifically (once you remove the mutex) the expect for the `UrlCleaner::clear_single_url_str` call should fallback to the input, instead of panicking. This shouldn't happen since the `clearurls::Error` [enum](https://docs.rs/clearurls/latest/clearurls/enum.Error.html) is pretty small (and default embedded rules presumably exclude most of them); but a malformed URL is very likely.